atecc608a/main.c

```diff
@@ -76,18 +76,32 @@
     "the slot now behaves according to the policies set by the associated\n"\
     "configuration zone’s values. [y/n]: "
 
+static void psa_purge_keys(void)
+{
+    psa_key_id_t id;
+    printf("Clearing persistent key data\n");
+    /* Clear slots 0-15 of persistent storage. These numbers are equal to
+     * physical slots available on the device. */
+    for (id = 0; id < 16; id++) {
+        psa_destroy_persistent_key(id);
+    }
+    /* Purge the transaction file. */
+    psa_crypto_stop_transaction();
+}
+
 /* Data used by tests */
-psa_key_slot_number_t atecc608a_private_key_slot = 0;
+psa_key_slot_number_t atecc608a_private_key_slot = 1;
 psa_key_slot_number_t atecc608a_public_key_slot = 9;
```

Comment:
If we got rid of these, we'd have to have a driver that implemented "allocate", right? Since this is non-ideal, we should leave a TODO comment to tell users "don't do this" and to remind us to fix and file an issue.

Reply:
In the example - yes. We would still need to have information about the hardware configuration somewhere. Ideally - the driver could read it in runtime and allocate a key, but that's a lot of work, and it's flaky (unless we make it bloated and check **a lot** of stuff).